### PR TITLE
Fix ta fetch

### PR DESCRIPTION
**Pull request by omuravskiy**

### CODE REVIEW

File: rpki-validator-app/src/main/scala/net/ripe/rpki/validator/models/validation/ValidationProcess.scala [hunk 1]
```diff
@@ -110,60 +110,59 @@ class TrustAnchorValidationProcess(override val trustAnchorLocator: TrustAnchorL
...
-      val taCertificate = matchingCertificates.head
-      ValidatedObject.valid(
-        Some("cert", taCertificate),
-        Lists.newArrayList(taCertificate.decoded.getSubject.getName),
-        taCertUri,
-        Some(taCertificate.hash),
-        (validationResult.getAllValidationChecksForCurrentLocation.asScala ++ fetchErrorChecks).toSet,
-        taCertificate.decoded
-      )
+    validCertificateObjects.headOption match {
```
Comment (lolepezy) [inline rpki-validator-app/src/main/scala/net/ripe/rpki/validator/models/validation/ValidationProcess.scala]: Should we simply take the headOption or have the check with 
ValidationString.VALIDATOR_REPOSITORY_TA_CERT_URI_NOT_UNIQUE ?

File: rpki-validator-app/src/main/scala/net/ripe/rpki/validator/rtr/Pdu.scala [hunk 2]
```diff
@@ -285,6 +285,7 @@ object Pdus {
   }
 
   private def parseErrorPdu(buffer: ChannelBuffer, headerShort: Short): Right[Nothing, ErrorPdu] = {
+    // NB: this will break for lengths between 2^31 and 2^32
```
Comment (lolepezy) [inline rpki-validator-app/src/main/scala/net/ripe/rpki/validator/rtr/Pdu.scala]: Shouldn't it be readUnsignedInt then?

Comment (omuravskiy) [reply rpki-validator-app/src/main/scala/net/ripe/rpki/validator/rtr/Pdu.scala]: This is a tricky bit. readUnsignedInt returns a long, with max value of 2^32-1. But you can't create pretty much anything in Java with such length, the max length of byte array is Integer.MAX_LENGTH, you can't even pass Long value as array's size.
I guess the proper solution is to maintain a list of arrays if the size is bigger than 2^31 :(

File: rpki-validator-app/src/main/scala/net/ripe/rpki/validator/models/validation/ValidationProcess.scala (hunk 1 above)
Comment (omuravskiy) [reply rpki-validator-app/src/main/scala/net/ripe/rpki/validator/models/validation/ValidationProcess.scala]: Yes, I'll submit an update with headOption map ... getOrElse ...
The VALIDATOR_REPOSITORY_TA_CERT_URI_NOT_UNIQUE should not be possible anymore with the new logic (with old logic it was possible to have objects with the same URI but different hashes _in the store_, but we do not use store at this point anymore).

**Pull request merged**